### DISCUSSION TITLE
API queue

### DISCUSSION THREAD

**Discussion by ghost** [oobabooga/text-generation-webui#1376] [Q&A]
I'm using the API extension. if I send multiple API generate requests at once, it will react wierd and ignore at least one of them and/or crash. Is this normal behaviour? There should probably be a queue or something for this

**Comment by danpizzy** [comment]
how did you get the api to work please  can i get your api code

**Comment by danpizzy** [comment]
how did you get the api to work please  can i get your api code

**Comment by danpizzy** [comment]
i keep getting jsondecode error no matter what i try

**Reply by ghost** [reply]
Please don't spam. 


I can provide some information:
 - Startup command i use: `python3 server.py --model eachadea_ggml-vicuna-13b-1.1 --listen --extensions api`
 - How i interact with the API (Javascript):
```js
import axios from 'axios';

const apiBase = "http://localhost:5000"
const context = "This is a conversation with your Assistant. The Assistant is very helpful and is eager to chat with you and answer your questions.\nAssistant: Hello! How can i help you?";
const stopping_strings = ["\nAssistant:", "\nYou:"];

const myPrompt = "Hello, what is 1 + 1?"

const reponse = await axios.post(`${apiBase}/api/v1/generate`, {
    prompt: `${context}\nYou: ${myPrompt}\nAssistant: `,
    stopping_strings: stopping_strings
}, {
    headers: {
        'Content-Type': 'application/json'
    }
});

const reponseText = reponse.data.results[0];

console.log(responseText.text); // OUTPUT

```

Im not using the gradio api, only the API EXTENTION

**Reply by danpizzy** [reply]
ok thank you very very much letme try that

**Reply by danpizzy** [reply]
i want to increase my max tokens

**Reply by harvard1932** [reply]
how come the API doesnt always reply back with a response? Only work like 1/20 of the time..

**Comment by danpizzy** [comment]
i love you bro i have been up all night trying to get this to work
by the way how do i add parameters to this

**Comment by danpizzy** [comment]
for anyone who has also been encountering the jsondecode issue with the default api example this is one that works for me 
import requests

url = "http://localhost:5000/api/v1/generate"

payload = "{\r\n    \"prompt\": \"This is a conversation with your Assistant. The Assistant is very helpful and is eager to chat with you and answer your questions.\\nYou: Hello, what is 1 + 1?\\nAssistant: \",\r\n    \"stopping_strings\": [\"\\nAssistant:\", \"\\nYou:\"]\r\n}\r\n"
headers = {
  'Content-Type': 'text/plain'
}

response = requests.request("POST", url, headers=headers, data=payload)

print(response.text)

**Reply by harvard1932** [reply]
this code doesnt always work, it only work less than 10% of the time
